sherpa/astro/xspec/tests/test_xspec.py

```diff
@@ -611,3 +575,138 @@ def __init__(self, name='zbabs'):
     expected = XSzbabs()([1, 2, 3])
 
     assert_array_equal(expected, actual)
+
+
+# The following repeats logic in self.assertFinite above.
+# The plan is to move the SherpaTestCase derived tests into
+# plain pytest-style tests, and then self.assertFinite
+# will be removed, but that day is not today.
+#
+# In Sherpa 4.8.0 development, the XSPEC model class included
+# an explicit check that all the bins were finite. This has
+# been removed as testing has shown there are complications when
+# using this with some real-world responses. So add in an
+# explicit - hopefully temporary - test here.
+#
+# Also ensure that at least one bin is > 0
+# (technically we could have a model with all negative
+#  values, or all 0 with the default values, but this
+#  seems unlikely; it is more likely a model evaluates
+#  to 0 - or at least not positive - because a data file
+#  is missing).
+#
+def assert_is_finite(vals, modelcls, label):
+    """modelcls is the model class (e.g. xspec.XSphabs)"""
+
+    import sherpa.astro.xspec as xs
+
+    emsg = "model {} is finite [{}]".format(modelcls, label)
+    assert numpy.isfinite(vals).all(), emsg
+
+    # XSPEC 12.10.0 defaults to ATOMDB version 3.0.7 but
+    # provides files for 3.0.9 (this is okay for the application
+    # as it is automatically over-written by the XSPEC init file,
+    # but not for the models-only build we use). With this
+    # mis-match, APEC-style models return all 0 values, so check
+    # for this.
+    #
+    # Some models seem to return 0's, so skip them for now.
+    #
+    # The *cflow models return 0's because:
+    #     XSVMCF: Require z > 0 for cooling flow models
+    # but the default value is 0 but mkcflox/vmcflow
+    # have a default redshift of 0 in XSPEC 12.10.0 model.dat
+    #
+    if modelcls in [xs.XSismabs, xs.XSkerrd, xs.XSmkcflow, xs.XSvmcflow]:
+        # perhaps should check that all values == 0 so that
```

Comment:
I agree with checking the zeroness of the returned value, so we would also check against pesky build issues. Although I have no idea how the specific models are implemented and whether what I said makes any sense for these models.